src/analysis/calculus/deriv.lean

```diff
@@ -1880,6 +1881,10 @@ p.differentiable.differentiable_on
 @[simp] protected lemma deriv : deriv (λx, p.eval x) x = p.derivative.eval x :=
 (p.has_deriv_at x).deriv
 
+@[simp] protected lemma deriv_aeval {R : Type*} [comm_semiring R] [algebra R 𝕜] (p : polynomial R) :
+deriv (λ (x : 𝕜), aeval x p) x = aeval x (derivative p) :=
+by simp [aeval_def, eval₂_eq_eval_map]
+
```

Comment:
I don't really want to add more to this PR at this point, but I should point out that there are 10-15 auxiliary lemmas here about `differentiable` and so on for `(λx, p.eval x)`. Maybe some of these should be replicated for aeval.

Reply:
#18945